mmengine/hooks/checkpoint_hook.py

```diff
@@ -304,6 +327,50 @@ def after_val_epoch(self, runner, metrics):
 
         self._save_best_checkpoint(runner, metrics)
 
+    def after_train(self, runner) -> None:
+        """Publish the checkpoint after training.
+
+        Args:
+            runner (Runner): The runner of the training process.
+        """
+        if self.published_keys is None:
+            return
+
+        if self.save_last:
+            last_ckpt = runner.message_hub.get_info('last_ckpt')
+            assert last_ckpt ('Did not find last_checkpoint to be resumed.')
+            self._publish_model(runner, last_ckpt)
+
+        if self.save_best is not None:
+            if not self.best_ckpt_path:
+                raise RuntimeError(xxx)
+            self._publish_model(runner, best_ckpt)
+
+    def _publish_model(self, runner, out_file: str) -> None:
+
+        if published_keys is None:
+            return
+        checkpoint = runner.load_checkpoint(out_file)
+        ckpt_keys = list(checkpoint.keys())
+        published_keys = self.published_keys
+        for k in ckpt_keys:
+            if k not in published_keys:
+                print_log(
+                    f'Key `{k}` will be removed because it is not in '
+                    f'save_keys. If you want to keep it, '
+                    f'please set `{k}` in published_keys',
+                    logger='current')
+                checkpoint.pop(k)
```

Comment:
```suggestion
        removed_keys = []
        for key in checkpoint.keys():
            if key not in self.published_keys:
                removed_keys.append(key)
                checkpoint.pop(k)
        if removed_keys:     
            print_log(
                f'Key {removed_keys} will be removed because they are not '
                'found in published_keys. If you want to keep them, '
                f'please set `{removed_keys}` in published_keys',
                logger='current')
```